python/cudf/cudf/tests/test_indexing.py

```diff
@@ -1037,6 +1037,7 @@ def test_series_setitem_string(key, value):
     [
         ("a", 4),
         ("b", 4),
+        ("d", 4),
```

Comment:
This should work for numeric indexes as well right? If so we might want to have some tests for that. Also I recommend adding a test that checks that the type coercion works as expected.